schema/validator.go

```diff
@@ -44,6 +50,13 @@ func (v Validator) Validate(src io.Reader) error {
 		return errors.Wrap(err, "unable to read the document file")
 	}
 
+	if f := mapValidateDescendants[v]; f != nil {
```

Comment:
I'm fine panicking if somebody inserts a `nil` value in `mapValidateDescendants`.  I would be happier if Go's typing was strict enough to error with `cannot convert nil to type validateDescendantsFunc` when they attempted the insertion, but I don't see a way to do that if the [zero value for function types is `nil`][1].  To be extra safe and clear, I'd recommend handling the three cases explicitly:

    f, ok := mapValidateDescendents[v]
    if ok {
        if f == nil {
            …internal error: shame on you for putting a nil value in mapValidateDescendents…
        }
        …validate descendants…
    }
[1]: https://golang.org/ref/spec#Function_types